From 695e60a8027660e902ab69e97d911dc862962c4a Mon Sep 17 00:00:00 2001
From: Sun Yuechi <sunyuechi@iscas.ac.cn>
Date: Wed, 16 Sep 2026 19:07:20 -0700
Subject: [PATCH] systemd: keep the FUSE mount of ceph-fuse@.service visible on
 the host

ProtectControlGroups and ProtectKernel{Logs,Modules,Tunables} give the
service a private mount namespace, which hides the mount made on %I.
Turn them off and keep their capability and syscall parts.

Signed-off-by: Sun Yuechi <sunyuechi@iscas.ac.cn>
---
 systemd/ceph-fuse@.service.in | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/systemd/ceph-fuse@.service.in b/systemd/ceph-fuse@.service.in
index add962c4018b..2a7dfc8f35d2 100644
--- a/systemd/ceph-fuse@.service.in
+++ b/systemd/ceph-fuse@.service.in
@@ -6,6 +6,7 @@ Conflicts=umount.target
 PartOf=ceph-fuse.target
 
 [Service]
+CapabilityBoundingSet=~CAP_SYS_MODULE CAP_SYSLOG
 EnvironmentFile=-@SYSTEMD_ENV_FILE@
 ExecStart=@CMAKE_INSTALL_PREFIX@/bin/ceph-fuse -f %I
 LockPersonality=true
@@ -13,15 +14,17 @@ MemoryDenyWriteExecute=true
 NoNewPrivileges=true
 # ceph-fuse requires access to /dev fuse device
 PrivateDevices=no
-ProtectControlGroups=true
+# a private mount namespace would hide the FUSE mount on %I from the host
+ProtectControlGroups=false
 ProtectHostname=true
-ProtectKernelLogs=true
-ProtectKernelModules=true
-ProtectKernelTunables=true
+ProtectKernelLogs=false
+ProtectKernelModules=false
+ProtectKernelTunables=false
 Restart=on-failure
 RestrictSUIDSGID=true
 StartLimitBurst=3
 StartLimitInterval=30min
+SystemCallFilter=~@module syslog
 TasksMax=infinity
 
 [Install]
